src/targets/gpu/kernels/include/migraphx/kernels/float8.hpp

```diff
@@ -0,0 +1,564 @@
+    {
+        return fp8e4m3fn(0x20, fp8e4m3fn::from_bits());
+    }
+    // NOLINTNEXTLINE
+    static constexpr __device__ fp8e4m3fn quiet_NaN()
+    {
+        return fp8e4m3fn(0x7F, fp8e4m3fn::from_bits());
+    }
+
+    static constexpr __device__ fp8e4m3fn max() { return fp8e4m3fn(0x7E, fp8e4m3fn::from_bits()); }
+    // this is min value that is not DeNorm. DeNorm min is 0x01
+    static constexpr __device__ fp8e4m3fn min() { return fp8e4m3fn(0x08, fp8e4m3fn::from_bits()); }
+
+    static constexpr __device__ fp8e4m3fn lowest()
+    {
+        return fp8e4m3fn(0xFE, fp8e4m3fn::from_bits());
+    }
+};
+
+template <>
+class numeric_limits<fp8e5m2fnuz>
+{
+    public:
+    static constexpr bool has_infinity = false;
+    static constexpr __device__ fp8e5m2fnuz epsilon()
+    {
+        return fp8e5m2fnuz(0x34, fp8e5m2fnuz::from_bits());
+    }
+
+    static constexpr __device__ fp8e5m2fnuz quiet_NaN() // NOLINT
+    {
+        return fp8e5m2fnuz(0x80, fp8e5m2fnuz::from_bits());
+    }
+
+    static constexpr __device__ fp8e5m2fnuz max()
+    {
+        return fp8e5m2fnuz(0x7F, fp8e5m2fnuz::from_bits());
+    }
+    // this is min value that is not DeNorm. DeNorm min is 0x01. I am not sure if we want to make
+    // this distinction. For the floating points we would end up using lowest most of the times.
+    static constexpr __device__ fp8e5m2fnuz min()
+    {
+        return fp8e5m2fnuz(0x4, fp8e5m2fnuz::from_bits());
+    }
+
+    static constexpr __device__ fp8e5m2fnuz lowest()
+    {
+        return fp8e5m2fnuz(0xFF, fp8e5m2fnuz::from_bits());
+    }
+};
+
+template <>
+class numeric_limits<fp8e5m2>
+{
+    public:
+    static constexpr bool has_infinity = true;
+    static constexpr __device__ fp8e5m2 epsilon() { return fp8e5m2(0x34, fp8e5m2::from_bits()); }
+    // 7D, 7E, 7F are positive NaNs and FD, FE, FF are negative NaNs
+    static constexpr __device__ fp8e5m2 quiet_NaN() // NOLINT
+    {
+        return fp8e5m2(0xFF, fp8e5m2::from_bits());
+    }
+
+    static constexpr __device__ fp8e5m2 max() { return fp8e5m2(0x7B, fp8e5m2::from_bits()); }
+    // this is min value that is not DeNorm. DeNorm min is 0x01. I am not sure if we want to make
+    // this distinction. For the floating points we would end up using lowest most of the times.
+    static constexpr __device__ fp8e5m2 min() { return fp8e5m2(0x4, fp8e5m2::from_bits()); }
+
+    static constexpr __device__ fp8e5m2 lowest() { return fp8e5m2(0xFB, fp8e5m2::from_bits()); }
+    // 7C and FC both are infinity
+    static constexpr __device__ fp8e5m2 infinity() { return fp8e5m2(0x7C, fp8e5m2::from_bits()); }
+};
+
+} // namespace fp8
+
+// NOLINTNEXTLINE
+#define MIGRAPHX_FP8_MIN_MAX(T)                  \
+    template <>                                  \
+    constexpr T numeric_max<T, void>()           \
+    {                                            \
+        return fp8::numeric_limits<T>::max();    \
+    }                                            \
```

Comment:
Doesn't work it, It would be considered as redefinition of `numeric_max()`. I need to speicalize `numeric_max` template that is defined in `type_traits.hpp`.

Reply:
Ok then you should be able to specialize it like this:

```cpp
template <migraphx::fp8::f8_type T = migraphx::fp8::f8_type::fp8, bool FNUZ, class Enable=void>
constexpr float8<T, FNUZ> numeric_max<float8<T, FNUZ>, Enable>()
{
    return fp8::numeric_limits<float8<T, FNUZ>>::max();
}
```

That should work for all fp8 types. Plus the `class Enable=void` wont require us to pass `void` into the function.

Reply:
Made changes.